### PR TITLE
Wip serialize xrp amount

### PR DESCRIPTION
**Pull request by joshwalong**

### CODE REVIEW

File: binary-codec/types/amount.go [hunk 1]
```diff
@@ -222,6 +203,31 @@ func VerifyIOUValue(value string) error {
...
 }
 
+// Serializes an XRP amount value
+func SerializeXrpAmount(value string) ([]byte, error) {
+
+	if VerifyXrpValue(value) != nil {
+		return nil, VerifyXrpValue(value)
+	}
+
+	bf, ok := new(big.Int).SetUint64(ZeroCurrencyAmountHex).SetString(value, 10)
```
Comment (JCrawsh) [inline binary-codec/types/amount.go]: This is a big int rather than a big float?

File: binary-codec/types/amount.go [hunk 2]
```diff
@@ -222,6 +203,31 @@ func VerifyIOUValue(value string) error {
...
+
+	if !ok {
+		return nil, errors.New("failed to convert string to big.Float")
+	}
+
+	val := bf.Uint64() // XRP values are represented as integers as drops and values - can't be negative
```
Comment (JCrawsh) [inline binary-codec/types/amount.go]: In fact do we need to cast to a big int at all? or can we just use a uint64?

File: binary-codec/types/amount.go [hunk 3]
```diff
@@ -233,44 +239,60 @@ func VerifyIOUValue(value string) error {
...
+
+// 	serial := uint64(ZeroCurrencyAmountHex)
+// 	if bigDecimal.Sign == "" {
+// 		serial |= uint64(PosSignBitMask) // if the sign is positive, set the sign bit to 1
+// 		serial |= uint64(exp+97) << 54   // if the exponent is positive, set the exponent bits to the exponent + 97
+// 		serial |= uint64(mantissa)       // last 54 bits are mantissa
 // 	}
 
-// 	return nil, nil
+// 	return []byte(new(big.Int).SetUint64(serial).Bytes()), nil
```
Comment (JCrawsh) [inline binary-codec/types/amount.go]: If all of this isn't needed just delete?

File: binary-codec/types/amount_test.go [hunk 4]
```diff
@@ -155,6 +155,65 @@ func TestVerifyIOUValue(t *testing.T) {
...
+			name:           "invalid xrp value - decimal",
+			input:          "125000708.0",
+			expectedOutput: nil,
+			expErr:         errors.New("XRP value must not contain a decimal"),
+		},
+		{
+			name:           "invalid xrp value - out of range",
+			input:          "983496373783750346983046704237509273095720937560927469729076023947673406",
+			expectedOutput: nil,
+			expErr:         errors.New("XRP value is an invalid XRP amount"),
```
Comment (JCrawsh) [inline binary-codec/types/amount_test.go:212]: Can we add some boundary tests? One just before the boundary, on the boundary(max value) and over the boundary?

Comment (joshwalong) [reply binary-codec/types/amount_test.go:212]: done thanks

**Review by JCrawsh — APPROVED** [review]
LGTM

**Review by bmurphy333 — COMMENTED** [review]
lgtm

**Review by bmurphy333 — APPROVED** [review]
